### PR TITLE
[5.8] Add `bindings` and `singletons` to the ServiceProvider

### PR DESCRIPTION
**Pull request by TBlindaruk**
- this PR is preaty the same as https://github.com/laravel/framework/pull/27565. We do not have any reason for checking if property exist, in cas if we could just get property values ;)

### CODE REVIEW

File: src/Illuminate/Support/ServiceProvider.php [hunk 1]
```diff
@@ -23,6 +23,16 @@ abstract class ServiceProvider
      */
     protected $defer = false;
 
+    /**
+     * @var array
+     */
+    public $bindings = [];
```
Comment (antonkomarev) [inline src/Illuminate/Support/ServiceProvider.php]: Maybe make it `protected`?

File: src/Illuminate/Support/ServiceProvider.php [hunk 2]
```diff
@@ -23,6 +23,16 @@ abstract class ServiceProvider
...
+
+    /**
+     * @var array
+     */
+    public $singletons = [];
```
Comment (antonkomarev) [inline src/Illuminate/Support/ServiceProvider.php]: Maybe make it `protected`?

File: src/Illuminate/Support/ServiceProvider.php (hunk 1 above)
Comment (TBlindaruk) [reply src/Illuminate/Support/ServiceProvider.php]: then we should add the getter

Comment (antonkomarev) [reply src/Illuminate/Support/ServiceProvider.php]: Yes, my mistake. Haven't checked that it's used outside of the base provider.

**Comment by taylorotwell** [comment]
I don't think I'm going to add anymore to the service provider for now.

**Pull request closed without merging**